library/std/src/sys/unix/ext/process.rs

```diff
@@ -319,3 +320,70 @@ impl IntoRawFd for process::ChildStderr {
 pub fn parent_id() -> u32 {
     crate::sys::os::getppid()
 }
+
+/// Converts a signal number to a string representation suitable for a human reader.
+///
+/// This does not necessarily produce the same output as the C `strsignal` function.
+/// Currently there are two deviations from the return value of `strsignal`:
+///
+///  * `signal_describe` does not translate the string according to the locale; C `strsignal`
+///    usually does.  Unfortunately a translated version is not currently available.
+///
+///  * `signal_describe` will include the signal abbrevation (eg, `SIGINT`) if
+///    the platform C library can provide that.
+///
+/// The precise form of the output should not be relied on.
+///
+/// # Examples
+///
+/// ```
+/// #![feature(unix_signal_strings)]
+/// use std::os::unix::process::signal_describe;
+/// assert!(signal_describe(9).contains("Killed"));
+/// ```
+#[unstable(feature = "unix_signal_strings", issue = "none")]
+pub fn signal_describe(sig: i32) -> String {
+    let mut buf = String::new();
+    os::signal_display(&mut buf, sig).unwrap();
+    buf
+}
+
+/// Looks up a signal number to get a basic string representation.
+///
+/// For known signals, this will typically give the same output as the C `strisignal`
+/// function does in the `C` locale.
+///
+/// Will return `None` for unknown or invalid signal numbers.
+///
+/// # Examples
+///
+/// ```
+/// #![feature(unix_signal_strings)]
+/// use std::os::unix::process::signal_string_raw;
+/// assert_eq!(signal_string_raw(15), Some("Terminated"));
+/// ```
+#[unstable(feature = "unix_signal_strings", issue = "none")]
+pub fn signal_string_raw(sig: i32) -> Option<&'static str> {
+    os::signal_lookup::descrs.lookup(sig)
+}
+
+/// Looks up a signal number to get its abbreviation.
+///
+/// When this succeeds, it will return the part of the signal name after `SIG`.
+///
+/// On some systems, `signal_abbrev` is not supported and will always return `None`.  Whether it is
+/// supported is not known at compile time; it can depend on the libc version found at runtime.
+///
+/// # Examples
+///
+/// ```
+/// #![feature(unix_signal_strings)]
+/// use std::os::unix::process::signal_abbrev;
+/// if let Some(got) = signal_abbrev(1) {
+///   assert_eq!(got, "HUP");
+/// }
+/// ```
+#[unstable(feature = "unix_signal_strings", issue = "none")]
+pub fn signal_abbrev(sig: i32) -> Option<&'static str> {
```

Comment:
naming nit: avoid abbreviating words.

Reply:
Also, is there a rule against abbreviating things?  `Ref` `ptr` `alloc` `iter` `mut` `vec` `env` `args` .... `SocketAddr` `Dir` .... etc. etc.

Reply:
>  my understanding is that the preference tends to lean towards names that are not abbreviated as far as the recent additions to the standard library API surface go.

Fair enough.  I think, though, that in this case the abbreviation is definitely justified both by the prior art in glibc (which feels very natural to me as an old Unix hand) and by the clumsiness of the unabbreviated word "abbreviation" :-).

If this is a blocker for this MR then I can of course respray this corner of the bikeshed.  In any case, thanks for your attention.

Reply:
By way of hopefully sidestepping the bikeshed about abbreviations, I think it'd be fair to call this the `signal_name`. I don't think we need to call it an "abbreviation" at all. (Yes, `TERM` is itself an abbreviation, but it's the actual name of the signal.) And it makes sense to say we're mapping a number to a "name" here.

Reply:
> ... I think it'd be fair to call this the `signal_name`. I don't think we need to call it an "abbreviation" at all. (Yes, `TERM` is itself an abbreviation, but it's the actual name of the signal.) And it makes sense to say we're mapping a number to a "name" here.

`name` is fine by me, so I will do that.  (And yes the BSDs even have this in the `sys_signame` array so that is precedent.)

> By way of hopefully sidestepping the bikeshed about abbreviations,  ...

:-).